source/RemoteWiring/HardwareProfile.cpp

```diff
@@ -0,0 +1,353 @@
+/*
+    Copyright(c) Microsoft Open Technologies, Inc. All rights reserved.
+
+    The MIT License(MIT)
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files(the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions :
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+*/
+
+#include "pch.h"
+#include "HardwareProfile.h"
+#include "RemoteDevice.h"
+
+using namespace Microsoft::Maker::Firmata;
+using namespace Microsoft::Maker::RemoteWiring;
+
+//******************************************************************************
+//* Constructors / Destructors
+//******************************************************************************
+
+HardwareProfile::HardwareProfile(
+    Windows::Storage::Streams::IBuffer ^buffer_,
+    Protocol protocol_
+    ) :
+    _is_valid( ATOMIC_VAR_INIT( false ) ),
+    _total_pins( ATOMIC_VAR_INIT( 0 ) ),
+    _analog_offset( ATOMIC_VAR_INIT( 0 ) ),
+    _num_analog_pins( ATOMIC_VAR_INIT( 0 ) ),
+    _pinCapabilities( nullptr ),
+    _analogResolutions( nullptr ),
+    _pwmResolutions( nullptr ),
+    _servoResolutions( nullptr )
+{
+    switch( protocol_ )
+    {
+    default:
```

Comment:
So in the event that they static cast an integer to `Protocol` then? I can't think of another case where that could really happen, but I suppose it would be throw worthy, since it is a programmer's mistake.